build: add "Build with Docker" guide (#17051)
docker/docs: build/guide/export.md

      go build -o /bin/client ./cmd/client

  FROM base as build-server
  ARG APP_VERSION="0.0.0+unknown"
  RUN --mount=type=cache,target=/go/pkg/mod/ \
      --mount=type=bind,target=. \
      go build -ldflags "-X main.version=$APP_VERSION" -o /bin/server ./cmd/server

  FROM scratch AS client
  COPY --from=build-client /bin/client /bin/
  ENTRYPOINT [ "/bin/client" ]

  FROM scratch AS server
  COPY --from=build-server /bin/server /bin/
  ENTRYPOINT [ "/bin/server" ]
+
+ FROM scratch AS binaries
+ COPY --from=build-client /bin/client /
+ COPY --from=build-server /bin/server /
```

Now you can build the `binaries` target using the `--output` option to export
both the client and server binaries.

```console
$ docker build --output=bin --target=binaries .
$ ls -l ./bin
total 29392
-rwxr-xr-x  1 user  user  7581933 Apr  6 09:33 client
-rwxr-xr-x  1 user  user  7459368 Apr  6 09:33 server
```

## Summary

This section has demonstrated how you can use Docker to build and export
standalone binaries. These binaries can be distributed freely, and don’t require
a container runtime like the Docker daemon.

The binaries you've generated so far are Linux binaries. That's because the

> build environment is Linux. If your host OS is Linux, you’ll be able to run

Reviewer: ```suggestion
build environment is Linux. If your host OS is Linux, you can run
```
